lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -371,6 +378,15 @@ impl<'s> SegmentHolder {
                 )?
             };
             applied_points.insert(point_id);
+            #[cfg(debug_assertions)]
+            if is_applied {
+                // Check that the version of applied point is correct if present
+                let point_version = write_segment.point_version(point_id);
+                if let Some(point_version) = point_version {
+                    // version is either applied or the point was removed
+                    assert!(point_version >= op_num);
+                }
+            }
```

Comment:
It might be a bit of abstraction leak here, nobody forces us to update point version. But actually it happens everywhere. So I added this debug assertion to lower probability of misuse here.